core/app/(user)/magic-link/route.ts

```diff
@@ -16,30 +17,8 @@ const redirectToURL = (
 		searchParams?: Record<string, string>;
 	}
 ) => {
-	// it's a full url, just redirect them there
-	if (URL.canParse(redirectTo)) {
-		const url = new URL(redirectTo);
-		Object.entries(opts?.searchParams ?? {}).forEach(([key, value]) => {
-			url.searchParams.append(key, value);
-		});
-
-		return NextResponse.redirect(url, opts);
-	}
-
-	if (URL.canParse(redirectTo, env.PUBPUB_URL)) {
-		const url = new URL(redirectTo, env.PUBPUB_URL);
-
-		Object.entries(opts?.searchParams ?? {}).forEach(([key, value]) => {
-			url.searchParams.append(key, value);
-		});
-		return NextResponse.redirect(url, opts);
-	}
-
-	// invalid redirectTo, redirect to not-found
-	return NextResponse.redirect(
-		new URL(`/not-found?from=${encodeURIComponent(redirectTo)}`, env.PUBPUB_URL),
-		opts
-	);
+	const url = createRedirectUrl(redirectTo, opts?.searchParams);
```

Comment:
Moved this to a helper file since I found I needed it elsewhere (mostly to easily tack on search params). but I think @tefkah 's lil routes lib would be a lot nicer